src/main/java/org/broadinstitute/hellbender/utils/mcmc/AbstractParameterizedState.java

```diff
@@ -0,0 +1,127 @@
+package org.broadinstitute.hellbender.utils.mcmc;
+
+import org.broadinstitute.hellbender.utils.Utils;
+
+import java.util.*;
+
+/**
+ * Represents a collection of named {@link Parameter} objects, which may hold values of mixed type.
+ * See GibbsSamplerSingleGaussianUnitTest and GibbsSamplerCopyRatioUnitTest for examples of use.
+ *
+ * @author Samuel Lee &lt;slee@broadinstitute.org&gt;
+ */
+public abstract class AbstractParameterizedState {
+    private final Map<String, Parameter<?>> parameterMap = new HashMap<>();
+
+    /**
+     * Constructs an AbstractParameterizedState from a List of {@alink Parameter} objects with values of mixed type.
+     * @param parameters    List of Parameters
+     * @throws IllegalArgumentException if {@code parameters} is null or contains duplicate parameter names
+     */
+    public AbstractParameterizedState(final List<Parameter<?>> parameters) {
```

Comment:
Is there an MCMC state that is *not* parameterized?  If not, I feel `AbstractMCMCState` etc would be a more evocative name for this class.